File: CS/CustomReportClass.Module.Win/CustomReportWizardController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.Xpo;
using DevExpress.ExpressApp.Reports;
using DevExpress.ExpressApp.Reports.Win;

namespace CustomReportClass.Module.Win {    
    public partial class ReportWizardModifyController : WindowController {
        public ReportWizardModifyController() { }

        protected override void OnActivated() {
            base.OnActivated();

            WinReportServiceController reportServiceController = 
                Frame.GetController<WinReportServiceController>();
            reportServiceController.NewXafReportWizardShowing += 
                delegate(object sender, NewXafReportWizardShowingEventArgs e) {
                if(!e.ReportDataType.Equals(typeof(CustomReportData))) return;
                CustomReportWizardParameters newReportParamsObject = new 
                    CustomReportWizardParameters(e.WizardParameters.Report);
                newReportParamsObject.ReportName = "Custom report name";
                newReportParamsObject.AdditionalInfo = "AdditionalInfo default value";
                e.WizardParameters = newReportParamsObject;
            };
        }
    }

    [NonPersistent]
    internal class CustomReportWizardParameters : NewXafReportWizardParameters {
        public CustomReportWizardParameters(XafReport report) : base(report) { }

        public string AdditionalInfo {
            get { return ((CustomXafReport)Report).AdditionalInfo; }
            set { ((CustomXafReport)Report).AdditionalInfo = value; }            
        }       
    }
}

File: VB/CustomReportClass.Module.Win/CustomReportWizardController.vb
Imports System.ComponentModel
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp.Reports
Imports DevExpress.ExpressApp.Reports.Win
Imports DevExpress.Persistent.Validation

Namespace CustomReportClass.Module.Win
    Partial Public Class ReportWizardModifyController
        Inherits ViewController

        Private reportServiceController As WinReportServiceController
        Public Sub New()
        End Sub
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            reportServiceController = Frame.GetController(Of WinReportServiceController)()
            AddHandler reportServiceController.NewXafReportWizardShowing, AddressOf reportServiceController_NewXafReportWizardShowing
        End Sub
        Protected Overrides Sub OnDeactivated()
            RemoveHandler reportServiceController.NewXafReportWizardShowing, AddressOf reportServiceController_NewXafReportWizardShowing
            reportServiceController = Nothing
            MyBase.OnDeactivated()
        End Sub
        Private Sub reportServiceController_NewXafReportWizardShowing(ByVal sender As Object, ByVal e As NewXafReportWizardShowingEventArgs)
            If Not e.ReportDataType.Equals(GetType(CustomReportData)) Then
                Return
            End If
            Dim newReportParamsObject As New CustomReportWizardParameters(e.WizardParameters.Report)
            newReportParamsObject.ReportName = "Custom report name"
            newReportParamsObject.AdditionalInfo = "AdditionalInfo default value"
            e.WizardParameters = newReportParamsObject
        End Sub
    End Class
    <NonPersistent> _
    Public Class CustomReportWizardParameters
        Inherits NewXafReportWizardParameters

        Public Sub New(ByVal report As XafReport)
            MyBase.New(report)
        End Sub
        Public Property AdditionalInfo() As String
            Get
                Return CType(Report, CustomXafReport).AdditionalInfo
            End Get
            Set(ByVal value As String)
                CType(Report, CustomXafReport).AdditionalInfo = value
            End Set
        End Property
    End Class
End Namespace

